blacklight-cornell/spec/controllers/search_controller_spec.rb

```diff
@@ -0,0 +1,45 @@
+# spec/controllers/search_controller_spec.rb
+require "rails_helper"
+
+RSpec.describe SearchController, type: :controller do
+  describe "SearchController.transform_query" do
+    it "transforms the query correctly" do
+      query = "going fishing"
+      transformed_query = SearchController.transform_query(query)
+      expect(transformed_query).to eq('("going" AND "fishing") OR phrase:"going fishing"')
+    end
+
+    it "handles already-quoted queries correctly" do
+      # pending("Pending implementation of handling already-quoted queries")
+      query = '"going fishing"'
+      transformed_query = SearchController.transform_query(query)
+      expect(transformed_query).to eq("\"going fishing\"")
+    end
+
+    it "handles single-term queries correctly" do
+      query = "fishing"
+      transformed_query = SearchController.transform_query(query)
+      expect(transformed_query).to eq('("fishing") OR phrase:"fishing"')
+    end
+
+    it "handles apostrophised queries correctly" do
+      query = "a doll's house"
+      transformed_query = SearchController.transform_query(query)
+      expect(transformed_query).to eq('("a" AND "doll\'s" AND "house") OR phrase:"a doll\'s house"')
+    end
+
+    it "handles embedded quoted queries correctly" do
+      # pending("Pending implementation of handling embedded quoted queries")
+      query = "A \"fish finder\" going fishing offshore"
+      transformed_query = SearchController.transform_query(query)
+      expect(transformed_query).to eq("A \"fish finder\" going fishing offshore")
```

Comment:
Not sure if this transform_query works just for the d&a solr index, but in case it does:

Simple search and both current and NEW/upcoming ([DACCESS-360 branch](https://github.com/cul-it/blacklight-cornell/pull/2182)) advanced search all handle this quite differently.

Current simple search:
`(+A +quoted:"fish finder" +going +fishing +offshore)`
This is wrong: all words should be wrapped in quotes, we're switching to boolean ops instead of prefix ops, and this doesn't query across phrase fields at all for the non-quoted strings.

Current advanced search:
`( ((+"A" +""fish" +"finder"" +"going" +"fishing" +"offshore" ) OR (phrase:"A "fish finder" going fishing offshore")) )`
This is wrong: doesn't handle the quoted "fish finder" at all.

[NEW/upcoming advanced search](https://github.com/cul-it/blacklight-cornell/pull/2182):
`((("A") OR phrase:"A") AND (quoted:"fish finder") AND (("going" AND "fishing" AND "offshore") OR phrase:"going fishing offshore"))`
New advanced search handles strings with multiple quoted sections.

Reply:
I will add advanced search versions of these tests to document the behavior, right or wrong.